login_server/src/cpp/SingletonManager/EmailManager.cpp

```diff
@@ -141,6 +141,33 @@ int EmailManager::ThreadFunction()
 				mailClientSession.sendMessage(mailMessage);
 					email_sended = true;
 				}
+				catch (Poco::Net::SSLConnectionUnexpectedlyClosedException& ex) {
+					// it is a mad idea to send an email if the email sending failed
+					// errors.sendErrorsAsEmail();
+					// better wait instead and try again, it seems that the strato mail server sometimes discard or connection
+					// MAGIC NUMBER: sleep time if email sending failed (ssl connection was discarded from mailserver)
+					// wait 5 minute for the next try
+					Poco::Thread::sleep(5 * 60 * 1000);
+					// reconnect to mailserver
+					mailClientSession.login();
+					try {
+						mailClientSession.startTLS(ServerConfig::g_SSL_CLient_Context);
+						mailClientSession.login(Poco::Net::SMTPClientSession::AUTH_LOGIN, mEmailAccount.username, mEmailAccount.password);
+					}
+					catch (Poco::Net::SSLException& ex) {
+						errors.addError(new ParamError(function_name, "ssl certificate error", ex.displayText()));
+						printf("[PrepareEmailTask] ssl certificate error: %s\nPlease make sure you have cacert.pem (CA/root certificates) next to binary from https://curl.haxx.se/docs/caextract.html\n", ex.displayText().data());
+						return -1;
+					}
+					// retry only if not to many retries are already tried
+					// MAGIC NUMBER: Retry count if email sending failed
+					if (email->checkResendCounter() < 4) {
```

Comment:
I would define a constant for this `MAX_EMAIL_RESEND_COUNT`. We use the convention to CAPSLOCK CONSTANTS in order to easily identify them.